File: Python/my_first_node.py
#!/usr/bin/env python3

import rclpy # biblioteca padrão do ros para o python
from rclpy.node import Node

def main(args = None):
    rclpy.init(args = args) # inicializa a comunicação com o ros
    node = Node("py_test") #1 - Nome do nó
    node.get_logger().info("Hello World") # printa algo no terminal
    rclpy.spin(node) # Faz com que o nó, continue "rodando", até pressionar "ctrl c". #1 nome do nó
    rclpy.shutdown() # esta normalmente é a linha final do nó

if __name__ == '__main__':
    main()

File: C++/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

int main(int argc, char **argv){
    rclcpp::init(argc, argv); // Inicializa a comunicação com o ros2
    auto node = std::make_shared<rclcpp::Node>("cpp_test"); // Basicamente em c++, todos os nós são criados como smart pointers. Em Node: #1 Nome do nó
    RCLCPP_INFO(node->get_logger(), "Hello World"); // Um macro onde: #1 método de info do nó x. 2# a mensagem
    rclcpp::spin(node);// Mantém o nó rodando
    rclcpp::shutdown(); // Finaliza a comunicação com o ros2
    return 0;
}
